### PR TITLE
Basic image and text selection in article view

### PR DESCRIPTION
**Pull request by stephanebisson**
https://phabricator.wikimedia.org/T291861
https://phabricator.wikimedia.org/T291865

* Refactored the article transformation to include some style from the API response
* Block-level text selection.
* Image selection

https://wikimedia.github.io/wikistories-prototype/article/#/

### CODE REVIEW

File: src/store/article.js [hunk 1]
```diff
@@ -1,3 +1,45 @@
...
+        img.classList.add('pcs-widen-image-override')
+        img.classList.add('pcs-lazy-load-image-loaded')
+        f.replaceChildren(img)
+      } else {
+        f.remove()
+      }
+    }
+  }
+}
+
```
Comment (hueitan) [inline src/store/article.js:42]: 💯 to this form, codes explain themselves.

File: src/views/Article.vue [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            }
+          } else {
+            const selectedText = document.querySelector( '.selected-text' )
+            if ( selectedText ) {
+              selectedText.classList.remove( 'selected-text' )
+              if (selectedText !== e.target) {
+                e.target.classList.add('selected-text')
+              }
+            } else {
+              e.target.classList.add( 'selected-text' )
```
Comment (hueitan) [inline src/views/Article.vue:54]: I can select the <figure /> as the text

**Review by hueitan — COMMENTED** [review]
I think this is a good start with basic functionality 👍

**Comment by hueitan** [comment]
UI is a bit different than the mobile html? beside the link.
image

File: src/views/Article.vue [hunk 3]
```diff
@@ -3,51 +3,75 @@
...
-        selectedText: null
+        selectedText: null,
+        showImages: false,
+        images: []
+      }
+    },
+    computed: {
+      ...mapGetters( ['currentArticle'] ),
+      articleImages: () => {
+        return Array.from(document.querySelector('.article').querySelectorAll(('img'))).map(img => {
```
Comment (hueitan) [inline src/views/Article.vue:36]: learning from another project, capture all img will always get some unexpected images, for example in `Catherine,%20Duchess%20of%20Cambridge`

Comment (stephanebisson) [reply src/views/Article.vue:36]: Yeah I saw that, I expect the media-list endpoint to be more accurate but for the prototype I thought this was acceptable. Will improve it if needed.

Comment (hueitan) [reply src/views/Article.vue:36]: make sense to me 👍

File: src/components/ArticleImages.vue [hunk 4]
```diff
@@ -0,0 +1,58 @@
+<template>
+    <div class="article-images">
+        <div class="article-images-header">
+            <span>Select an image</span>
+            <a @click="onSubmit(selectedImage)" v-if="selectedImage">Next</a>
```
[the remark this answers is not in the document]
Comment (stephanebisson) [reply src/components/ArticleImages.vue:5]: Can you file it in phab?

**Pull request merged**